backend/src/rss-feed/rss-feed.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import sources from './rss-feed-sources';
+import { RssFeed } from './rss-feed.entity';
+import { Repository } from 'typeorm';
+const Parser = require('rss-parser');
+const parser = new Parser();
+
+
+@Injectable()
+export class RssFeedService {
+
+    constructor(@InjectRepository(RssFeed)
+    private readonly rssFeedRepository: Repository<RssFeed>) {
+
+    }
+
+    async onModuleInit() {
+        await this.fetchAndSaveRssFeeds();
+    }
+
+    async fetchAndSaveRssFeeds() {
```

Comment:
I believe transaction is useful here.
Also, can you please handle error cases whenever you are making a remote call to database and other network services?

Reply:
Yes definetly